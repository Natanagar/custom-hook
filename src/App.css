.App {
  text-align: center;
  max-width: 1024px;
  margin: 0 auto;
}

.open-button {
  border: 0;
  background: lightgreen;
  color: black;
  padding: 10px 20px;
  border-radius: 10px;
  font-size: 21px;
  cursor: pointer;
  margin-top: 20px;
}
