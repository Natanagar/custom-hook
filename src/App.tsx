import React from "react";
import "./App.css";

import useModals from "./hooks/useModals";

function App() {
  const doSomething = (e: React.MouseEvent, id: string) => {
    alert(`Колбек выполнен на модальном окне с id = ${id}`);
  };
  const { Modals, createModal } = useModals(doSomething);

  return (
    <div className="App">
      <button className="open-button" onClick={createModal}>
        Создать окно
      </button>
      <Modals />
    </div>
  );
}

export default App;
