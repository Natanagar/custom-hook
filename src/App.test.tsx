import React from "react";
import renderer from "react-test-renderer";
import App from "./App";

describe("App renders without mistakes", () => {
  it("matches snapshot", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
