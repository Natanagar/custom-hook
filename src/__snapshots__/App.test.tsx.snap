// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without mistakes matches snapshot 1`] = `
<div
  className="App"
>
  <button
    className="open-button"
    onClick={[Function]}
  >
    Создать окно
  </button>
  <div
    className="modals"
  />
</div>
`;
